manifests/job.pp

```diff
@@ -18,6 +22,7 @@
 #
 define jenkins::job(
   $config,
+  $source = '',
```

Comment:
Why not make the default `undef`?  I suggest validating the param with `validate_string()`.